src/Microsoft.SqlTools.ServiceLayer/Agent/Jobs/JobStepData.cs

```diff
@@ -365,6 +365,12 @@ public StepCompletionAction FailureAction
                 }
                 return this.failureAction;
             }
+            set
+            {
+                if (this.failureAction != value) {
```

Comment:
Seems a little silly to have this check but I guess if that's the pattern I don't have a huge issue with it...